### PR TITLE
Implementation, with curl support.

### PR DESCRIPTION
**Pull request by jmjoy**
The Implementation of skywalking php agent, with the curl support. and the e2e test.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# limitations under the License.
+
+[package]
+name = "skywalking-php"
+version = "0.1.0"
+authors = ["Apache Software Foundation", "jmjoy <jmjoy@apache.org>", "Yanlong He <heyanlong@apache.org>"]
+description = "Apache SkyWalking PHP Agent."
+edition = "2021"
+rust-version = "1.58"
+repository = "https://github.com/jmjoy/skywalking-php"
```
Comment (heyanlong) [inline Cargo.toml:23]:
```suggestion
repository = "https://github.com/apache/skywalking-php"
```

File: Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+license = "MulanPSL-2.0"
```
Comment (heyanlong) [inline Cargo.toml:24]:
```suggestion
license = "Apache-2.0"
```

File: Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+readme = "README.md"
+publish = false
+
+[lib]
+name = "skywalking_agent"
```
Comment (heyanlong) [inline Cargo.toml:29]: Is it possible to use skywalking-agent?

Comment (jmjoy) [reply Cargo.toml:29]: The field `name` is not allow `-`.

File: src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+/// Log level of skywalking agent.
+const SKYWALKING_AGENT_LOG_LEVEL: &str = "skywalking_agent.log_level";
+
+/// Log file of skywalking agent.
+const SKYWALKING_AGENT_LOG_FILE: &str = "skywalking_agent.log_file";
+
+#[php_get_module]
+pub fn get_module() -> Module {
+    let mut module = Module::new(
+        env!("CARGO_CRATE_NAME"),
```
Comment (jmjoy) [inline src/lib.rs:61]: @heyanlong  Here, the extension name is refer to `Cargo.toml` field `lib.name`, is `skywalking_agent`.

**Comment by heyanlong** [comment]
LGTM

**Review by heyanlong — APPROVED** [review]

**Pull request merged**